### PR TITLE
Map frontend

### PR DESCRIPTION
**Pull request by gusteran**
- Add a page with with a google maps api
- Center the map on the US and add small styling to it
- Allow the user to enter a US zip code and see the superfund sites in their zip
- Add a button to view the superfund sites in the US state of the most recent zip code entered

### CODE REVIEW

**Comment by natalie-abreu** [comment]
Can you add a picture of what it looks like?

File: src/main/webapp/map.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            marker.addListener('click', () => {
+                    if(lastInfoWindow != null) {lastInfoWindow.close();}
+                    infoWindow.open(map, marker);
+                    lastInfoWindow = infoWindow;
+                });
+                map.addListener('click', () => {
+                    infoWindow.close();
+                });
+        });
+        console.log(sites);
```
Comment (natalie-abreu) [inline src/main/webapp/map.js:115]: Take out print statements

File: src/main/webapp/map.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            map.fitBounds(lastGeocode.geometry.viewport);
+            bounds.union(lastGeocode.geometry.viewport);
+            
+            addSuperfundMarkers(areaType, areaCode);
+
+        } else {
+
+            map.setZoom(4);
+            map.setCenter({lat: 42, lng: -93});
+            console.log("Reset map center and zoom");
```
Comment (natalie-abreu) [inline src/main/webapp/map.js:71]: Remove

File: src/main/webapp/map.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    if(lastGeocode === null) return;
+    var state = "01";
+    for(i in lastGeocode.address_components){
+        if(lastGeocode.address_components[i].short_name.length === 2){
+            state = lastGeocode.address_components[i].short_name;
+            break;
+        }
+    }
+    addSuperfundMarkers("state", state);
+    console.log("Zoom to state of "+state);
```
Comment (natalie-abreu) [inline src/main/webapp/map.js:86]: ^

File: src/main/webapp/map.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            marker.addListener('click', () => {
+                    if(lastInfoWindow != null) {lastInfoWindow.close();}
+                    infoWindow.open(map, marker);
+                    lastInfoWindow = infoWindow;
+                });
+                map.addListener('click', () => {
+                    infoWindow.close();
+                });
+        });
+        console.log("Placed Superfund Markers");
```
Comment (natalie-abreu) [inline src/main/webapp/map.js:115]: ^

**Comment by gusteran** [comment]
After entering my zip code
image
After clicking show state
image
Clicking on a marker currently shows this (it's name and locality)
image

**Comment by gusteran** [comment]
I have the console.logs that I still do as I feel them important logs to let me know that components have worked (ie. the geocoding response or the superfund servlet response)

**Comment by natalie-abreu** [comment]
Does view state require the zip code put in first (like a zoom out to state option) or is it geolocating the user's current state? Maybe make the button label more descriptive

**Comment by gusteran** [comment]
Now shows which state will be zoomed to on the button label
image

File: src/main/java/com/google/sps/servlets/SuperfundServlet.java [hunk 5]
```diff
@@ -50,6 +50,8 @@
 
     private static final String ZIP_PARAMETER = "zip_code";
 
+    public static final int DEFAULT_SCORE = 100;
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/servlets/SuperfundServlet.java:53]: The name score is generic. What type of score is this?

File: src/main/webapp/map.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    }).then((sites) => {
+        map.fitBounds(bounds);
+    });
+}
+
+function iconUrl(score){
+	var url = "https://maps.google.com/mapfiles/ms/icons/";
+	if(score < 25) url+= "green";
+	else if(score <40) url += "yellow";
+	else if (score < 55) url +="orange";
```
Comment (kavinkarthik92) [inline src/main/webapp/map.js:130]: This is core business logic. These numbers should be moved to static const

**Pull request merged**